### PR TITLE
Release v3.0.0 - Develop

### PR DESCRIPTION
**Pull request by bhazuka**
## Summary of Changes

Release 3.0.0 See CHANGELOG for full details.

### CODE REVIEW

File: tasks/request_files/request_files.py [hunk 1]
```diff
@@ -203,62 +228,127 @@ def inner_task(
...
+            glacier_bucket,
+            restore_expire_days,
+            max_retries,
+            retry_sleep_secs,
+            retrieval_type,
+            job_id,
+            db_queue_url,
+        )
+
+        # If no errors add to copied_granules array for output.
```
Comment (AndrewDorn) [inline tasks/request_files/request_files.py:342]: I might remove the "If no errors" since that isn't really a factor beyond as much as it is for any line.

Comment (AndrewDorn) [reply tasks/request_files/request_files.py:342]: Did you mean to have a try/except around process_granule?

File: tasks/request_files/request_files.py [hunk 2]
```diff
@@ -301,89 +391,109 @@ def process_granule(
...
+    # Try to restore objects in S3
     while attempt <= max_retries + 1:
         for a_file in granule[GRANULE_RECOVER_FILES_KEY]:
+            # Only do files we have not done or have not successfully been restored
             if not a_file[FILE_SUCCESS_KEY]:
                 try:
                     restore_object(
                         s3,
-                        a_file[FILE_KEY_KEY],
+                        a_file["key_path"],
```
Comment (AndrewDorn) [inline tasks/request_files/request_files.py:402]: Does not match input.json

File: tasks/request_files/request_files.py [hunk 3]
```diff
@@ -203,62 +228,127 @@ def inner_task(
...
         glacier_bucket = event[EVENT_CONFIG_KEY][CONFIG_GLACIER_BUCKET_KEY]
     except KeyError:
         raise RestoreRequestError(
             f"request: {event} does not contain a config value for glacier-bucket"
         )
 
+    # Get the granule array from the event
     granules = event[EVENT_INPUT_KEY][INPUT_GRANULES_KEY]
     if len(granules) > 1:
         # todo: This is either a lie, or the loop below should be removed.  ORCA-202
```
Comment (AndrewDorn) [inline tasks/request_files/request_files.py:242]: Looks like you've resolved everything having to do with ORCA-202

**Review by AndrewDorn — CHANGES_REQUESTED** [review]

File: tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py [hunk 4]
```diff
@@ -63,39 +64,51 @@ def tearDown(self):
...
         },
         clear=True,
     )
-    @patch("database.single_query")
-    @patch("requests_db.get_dbconnect_info")
+    @patch("post_copy_request_to_queue.shared_db.get_user_connection")
+    @patch("post_copy_request_to_queue.shared_db.get_configuration")
     @patch("post_copy_request_to_queue.shared_recovery.post_entry_to_queue")
     @patch("post_copy_request_to_queue.shared_recovery.update_status_for_file")
+    @patch("post_copy_request_to_queue.get_metadata_sql")  # todo: Unit test this sql?
```
Comment (AndrewDorn) [inline tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py:75]: We probably want a basic happy-path test for this function now that SQLAlchemy makes checking the raw string harder.

File: tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py [hunk 5]
```diff
@@ -63,39 +64,51 @@ def tearDown(self):
...
+        mock_get_user_connection.return_value = mock_engine
+
         record = self.event["Records"][0]
         new_data = {
             "job_id": "1",
             "granule_id": "3",
             "filename": "f1.doc",
-            "source_key": "b21b84d653bb07b05b1e6b33684dc11b",
-            "target_key": "b21b84d653bb07b05b1e6b33684dc11b",
+            # todo: Value incorrect here and elsewhere. As written, must match mock_execute's return value.
```
Comment (AndrewDorn) [inline tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py:108]: Looking now, I don't see any other issues where the restore_destination and mock_execute have different values.

File: tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py [hunk 6]
```diff
@@ -210,23 +212,23 @@ def test_event_non_int(self):
...
-        mock_LOGGER: MagicMock,
-        mock_post_entry_to_queue: MagicMock,
-        mock_update_status_for_file: MagicMock,
-        mock_get_dbconnect_info: MagicMock,
-        mock_single_query: MagicMock,
+            self,
+            mock_LOGGER: MagicMock,
+            mock_post_entry_to_queue: MagicMock,
+            mock_update_status_for_file: MagicMock,
+            mock_get_dbconnect_info: MagicMock,
```
Comment (AndrewDorn) [inline tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py:225]: 2 mislabeled mocks. Make the same changes here as in happy-path.

File: tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py [hunk 7]
```diff
@@ -252,26 +254,31 @@ def test_task_post_entry_to_queue_exception(
...
     def test_task_update_status_for_file_exception(
-        self,
-        mock_LOGGER: MagicMock,
-        mock_update_status_for_file: MagicMock,
-        mock_get_dbconnect_info: MagicMock,
-        mock_single_query: MagicMock,
+            self,
+            mock_LOGGER: MagicMock,
+            mock_update_status_for_file: MagicMock,
+            mock_get_dbconnect_info: MagicMock,
```
Comment (AndrewDorn) [inline tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py:275]: 2 mislabeled mocks. Make the same changes here as in happy-path.

**Review by AndrewDorn — CHANGES_REQUESTED** [review]
Some more test issues. My guess is that some tests are failing/would fail for the wrong reasons.

**Review by rizbi-hassan — APPROVED** [review]

**Comment by bhazuka** [comment]
Closed and reopening because of issues with Snyk

**Pull request closed without merging**